R/classification.R

```diff
@@ -0,0 +1,195 @@
+#' Retrieve the taxonomic hierarchies from local database
+#'
+#' This function is equivalent to the taxize::classification function, except
+#' that it uses a local database (so is much faster) and is currently
+#' restricted to handling NCBI taxa. The output is identical to
+#' taxize::classification.
+#'
+#' @param x Vector of taxon keys for the given database
+#' @param db The database to search
+#' @param verbose Print verbose messages
+#' @param ... Additional arguments passed to database specific classification functions.
+#' @return list of data.frames with the columns: name, rank, and id. This is
+#' exactly equivalent to the output of 'taxize::classification'. 
+#' @export
+#' @examples
+#' \dontrun{
+#' classification(c(3702, 9606))
+#' }
+classification <- function(x, db='ncbi', verbose=TRUE, ...){
+  lineages <- if(is.null(x) || length(x) == 0){
+    # For empty or NULL input, return empty list
+    list()
+  } else {
+    FUN <- switch(db,
+      itis = itis_classification,
+      tpl  = tpl_classification,
+      col  = col_classification,
+      gbif = gbif_classification,
+      ncbi = ncbi_classification
```

Comment:
couldn't we put `stop("Database '", db, "' is not supported")` as the last option in this switch statement?

Reply:
Yeah, that would be cleaner. I'll fix it in a second.